GeeksCoreLibrary/Modules/HealthChecks/Services/DatabaseHealthService.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using GeeksCoreLibrary.Core.Models;
+using GeeksCoreLibrary.Modules.Databases.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+
+namespace GeeksCoreLibrary.Modules.HealthChecks.Services;
+public class DatabaseHealthService : IHealthCheck
+{
+    private readonly IDatabaseConnection databaseConnection;
+    private readonly HttpContext httpContext;
+    private readonly HealthChecksSettings healthChecksSettings;
+    
+    public DatabaseHealthService(IDatabaseConnection databaseConnection, IHttpContextAccessor httpContextAccessor, IOptions<HealthChecksSettings> healthChecksSettings)
+    {
+        this.databaseConnection = databaseConnection;
+        httpContext = httpContextAccessor.HttpContext;
+        this.healthChecksSettings = healthChecksSettings.Value;
+    }
+    
+    /// <inheritdoc />
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context,
+        CancellationToken cancellationToken = new CancellationToken())
+    {
+        databaseConnection.ClearParameters();
+       
+        // A new query to check the max active connections from the DatabaseHealthCheck
+        var query =
+            $"SELECT COUNT(*) AS active_connections FROM information_schema.PROCESSLIST;";
+        var datatable = await databaseConnection.GetAsync(query);
+
+        if (datatable.Rows.Count == 0)
+        {
+            return await Task.FromResult(HealthCheckResult.Unhealthy("No data found"));
+        }
+         
+        var healthCheckConnections = healthChecksSettings.MaximumDatabaseConnections;
+        var healthCheckConnectionsTime = healthChecksSettings.MaximumConnectionsInTime;
+
+        // If no value is set, we are skipping this test.
+        if (healthCheckConnections > 0)
+        {
+            // Retrieve the count of active connections.
+            var activeConnections = Convert.ToInt32(datatable.Rows[0]["active_connections"]); 
+            
+            // Check if the number of active connections exceeds the limit.
+            if (activeConnections > healthCheckConnections) 
+            {
+                return await Task.FromResult(HealthCheckResult.Unhealthy(activeConnections +" databases connected, Too many database connections"));
+            }
+        }
+        
+        // If no value is set, we are skipping this test.
+        if (healthCheckConnectionsTime > 0)
+        {    
+            // Query to check the max open connections in time from the DatabaseHealthCheck.
+            query =  $"SELECT ID, USER, HOST, TIME AS connection_time_in_sec FROM information_schema.PROCESSLIST";
```

Comment:
You don't use any variables in this string, so please remove the `$` infront of it.

Also, you should add a filter to this query, because our databases have some connections that are open permanently. If you don't filter those out, this code will always return the unhealthy status, because they will almost always be higher than our threshold. One thing I know is that you should filter out all rows where the user is `repluser`, but there might be more, we'd have to look that up.

Reply:
added that user, let talk about more in v1.1 :)